streams/jvm/src/main/scala/zio/stream/platform.scala

```diff
@@ -289,27 +286,51 @@ trait ZStreamPlatformSpecificConstructors { self: ZStream.type =>
   /**
    * Creates a stream from [[java.io.Reader]].
    */
-  def fromReader(reader: => Reader): ZStream[Blocking, IOException, Char] =
-    ZStream.repeatEffectOption {
-      blocking
-        .effectBlockingIO(reader.read())
-        .foldM(
-          e => ZIO.fail(Some(e)),
-          r => if (r == -1) ZIO.fail(None) else ZIO.succeed(r.toChar)
-        )
+  def fromReader(reader: => Reader, chunkSize: Int = ZStream.DefaultChunkSize): ZStream[Blocking, IOException, Char] =
```

Comment:
I liked your previous approach with `repeatEffectOption` better :-) How about:
`ZStream.fromEffect(reader).flatMap { capturedReader => ZStream.repeatEffectOptionChunk(...)`?

Reply:
Haha, merged it, but dropped the exceptions used in first try.

Reply:
Btw. should I do the same for `fromInputStream`?

Reply:
Yeah why not. Can you add a few tests for `fromReader`?

Reply:
Done.